tutorials/cloudbuild-angular-universal/index.md

```diff
@@ -0,0 +1,301 @@
+```
+curl -LO https://raw.githubusercontent.com/GoogleCloudPlatform/community/master/tutorials/cloudbuild-angular-universal/webpack.prerender.config.js
+```
+
+2. Add the prerender webpack configuration file to git.
+```
+git add webpack.prerender.config.js
+```
+3. Download the typescript config file for prerendering.
+```
+curl -LO https://raw.githubusercontent.com/GoogleCloudPlatform/community/master/tutorials/cloudbuild-angular-universal/prerender.tsconfig.json
+```
+4. Add the  typescript config file for prerendering.
+```
+git add prerender.tsconfig.json
+```
+5. Download the prerender typescript file.
+```
+curl -LO https://raw.githubusercontent.com/GoogleCloudPlatform/community/master/tutorials/cloudbuild-angular-universal/prerender.ts
+```
+6. Add the prerender typscript file to git
+```
+git add prerender.ts
+```
+
+7. Modify the package.json file to add the prerender steps.
+
+**Note** that jq is a tool for editing JSON and is installed in Cloudshell by default. If you are going through this tutorial on your workstation see [jq installation](https://stedolan.github.io/jq/download/) for instructions on installing jq on your workstation.
+
+```bash
+read -r -d '' SCRIPT_ADDITIONS <<EOF
+{
+"build:prerender": "npm run build:client-and-server-bundles && npm run compile:prerender && npm run generate:prerender",
+"generate:prerender": "npm run webpack:prerender && node dist/prerender.js",
+"compile:prerender": "tsc -p prerender.tsconfig.json",
+"webpack:prerender": "webpack --config webpack.prerender.config.js"
+}
+EOF
+cat package.json | jq --argjson additions "$SCRIPT_ADDITIONS" '.scripts = .scripts+$additions' >tmpfile
+cp tmpfile package.json
+rm tmpfile
+```
+
+8. Add the package.json changes to git.
+```
+git add package.json
+```
+9. Commit your changes to the git repo
+```
+git commit -m "pregenerate changes"
+```
+### Configure a Cloud Storage Bucket and Cloud Load Balancer to host your Angular Application on Cloud CDN
+
+1. Create the content cloud storage bucket.
+```
+gsutil mb gs://$PROJECT-angular-app
+```
+2. Create the backend-bucket.
+```
+gcloud compute backend-buckets create $PROJECT-angular-app-backend \
+--gcs-bucket-name=$PROJECT-angular-app \
+--enable-cdn
+```
+3. Create a multi-regional ip address.
+```
+gcloud compute addresses create angular-app-ip --global
+ANGULAR_APP_IP=$(gcloud compute addresses list  --filter="name=angular-app-ip" --format="value(address)")
+```
+3. Create the url map.
+```
+gcloud compute url-maps create web-map --default-backend-bucket $PROJECT-angular-app-backend
+```
+4. Create the http proxy.
+```
+gcloud compute target-http-proxies create http-lb-proxy \
+--url-map web-map
+```
+5. Create the forwarding rule.
+```
+gcloud compute forwarding-rules create http-content-rule \
+--address angular-app-ip \
+--global \
+--target-http-proxy http-lb-proxy \
+--ports 80
+```
+## Create the Cloudbuild file and add it to the git repsoitory
+1. Give the cloudbuild cloud storage admin access.
```

Comment:
'Give the cloudbuild cloud storage admin access.'
->
'Give the Cloud Build account Cloud Storage admin access.'